pkg/osquery/tables/execparsers/repcli/parser.go

```diff
@@ -0,0 +1,171 @@
+			"service_uptime":              "",
+			"service_waketime":            "",
+		},
+		"cloud_status": {
+			"registered":         "",
+			"server_address":     "",
+			"next_check-in":      "",
+			"private_logging":    "",
+			"next_cloud_upgrade": "",
+			"mdm_device_id":      "",
+			"platform_type":      "",
+			"proxy":              "",
+		},
+		"proxy_settings": {
+			"proxy_configured": "",
+		},
+		"enforcement_status": {
+			"execution_blocks":     "",
+			"network_restrictions": "",
+		},
+		"rules_status": {
+			"policy_name":               "",
+			"policy_timestamp":          "",
+			"endpoint_standard_product": "",
+			"enterprise_edr_product":    "",
+			"active_policies":           make(map[string]string, 0),
+		},
+	}
+}
+
+// formatKey prepares raw (potentially multi-word) key values by:
+// - stripping all surrounding whitespace
+// - coercing the entire string to lowercase
+// - splitting multiple words and joining them as snake_case
+func formatKey(key string) string {
+	processed := strings.TrimSpace(strings.ToLower(key))
+	words := strings.Fields(processed)
+	// State and Status are used interchangeably in section headers
+	// across device types, so we coerce all to status here
+	for idx, word := range words {
+		if word == "state" {
+			words[idx] = "status"
+		}
+	}
+
+	return strings.Join(words, "_")
+}
+
+func repcliParse(reader io.Reader) (any, error) {
+	results := NewStatusTemplate()
+	var sectionHeaders []string
+	nestedLevel := 0
+
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		rawLine := scanner.Text()
+		if len(rawLine) == 0 {
+			continue // blank lines are not expected or meaningful
+		}
+
+		kv := strings.SplitN(rawLine, ":", 2)
+		if len(kv) < 2 {
+			continue // lines without a colon are not expected or meaningful
+		}
+
+		formattedKey := formatKey(kv[0])
+		formattedValue := strings.TrimSpace(kv[1])
+		isSectionHeader := len(formattedValue) == 0
+		// a nested section header has left padding and no associated value
+		currentNestedLevel := len(kv[0]) - len(strings.TrimLeftFunc(kv[0], unicode.IsSpace))
+		if isSectionHeader && currentNestedLevel <= 0 {
+			// reset key paths for any new, top-level header
+			sectionHeaders = []string{}
+		}
+
+		if isSectionHeader {
+			nestedLevel = currentNestedLevel
+			sectionHeaders = append(sectionHeaders, formattedKey)
+			continue
+		}
+		// from this point forward we expect that we're working with a
+		// line containing a full key-value pair. the following logic
+		// supports up to 2 levels of nesting
+		if len(sectionHeaders) > 2 || len(sectionHeaders) < 1 {
+			continue // log?
+		}
+
+		// if this is the case then we're leaving a nested section (and should
+		// re-enter the parent section instead)
+		if currentNestedLevel < nestedLevel && len(sectionHeaders) > 1 {
+			sectionHeaders = sectionHeaders[0 : len(sectionHeaders)-1]
+		}
+
+		if _, ok := results[sectionHeaders[0]]; !ok {
+			continue // we don't handle this section
+		}
```

Comment:
I think this has a lot of assumptions about the data. I think this should create the `map[string]any` if `!ok`. Might also need recursion

Reply:
I could maybe be pushed into this, but I've seen too many third parties change their data out from under us.